cmd/gen-go-sample/inittree.go

```diff
@@ -301,6 +351,33 @@ func (t *initTree) parsePathRest(sc *scanner.Scanner, info pbinfo.Info) (*initTr
 
 			t = t.index(indVal)
 
+		case '{':
+			isMap, err := t.typ.isMap()
+			if err != nil {
+				return nil, r, err
+			}
+			if !isMap {
+				return nil, r, errors.E(nil, "cannot index via {} into a non-map field")
+			}
+
+			var tokVal string
+			if r := sc.Scan(); r == scanner.Int || r == scanner.String || r == scanner.Ident {
+				// Note we don't yet support use a locally defined a variable (by the `define:` directive)
+				// as a map key. scanner.Ident is here to handle the case when key is a bool value, as
+				// `true` and `false` (without quote) are parsed as scanner.Ident.
+				tokVal = sc.TokenText()
+				validPrim := validPrims[t.typ.keyType.prim]
```

Comment:
See my responses on the other comment thread about this naming. These are two different types that differ by just one letter. We should fix this in a follow-up PR.

Reply:
Issue filed: https://github.com/googleapis/gapic-generator-go/issues/190